CHANGELOG.md

```diff
@@ -4,6 +4,80 @@
 
 ## DFX
 
+### feat: dfx extension can now be installed from external registries
+
+Modifies the `dfx extension install` command to take a flag `--registry` that allows installing extensions from external registries. Here is an example:
+
+```bash
+dfx extension install your_extension --registry=https://raw.githubusercontent.com/your-org/your-repo/master/dfx-extensions-registry.json
+```
+
+The registry file is a JSON file that contains a list of extensions. Here is an example:
+
+```json
+{
+  "compatibility": {
+    "0.15.0": {
+      "your_extension": {
+        "versions": ["0.1.0"]
+      }
+    }
+  },
+  "extensions": {
+    "your_extension": {
+      "0.1.0": {
+        "homepage": "https://github.com/your-org/your-repo",
+        "authors": "Your Org",
+        "summary": "Your extension",
+        "categories": ["development"],
+        "keywords": ["cli-helper"],
+        "description": "A longer description.",
+        "subcommands": {
+          "do_something": {
+            "about": "does something",
+            "args": {
+              "the_param": {
+                "about": "some paramater",
+                "long": "the-param"
+              }
+            }
+          }
+        },
+        "binaries": {
+          "unknown-linux-gnu-x86_64": {
+            "url": "https://raw.githubusercontent.com/your-org/your-repo/master/your_extension-v0.1.0-x86_64-unknown-linux-gnu.tar.gz",
+            "sha256": "abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890"
+          },
+          "apple-darwin-x86_64": {
+            "url": "https://raw.githubusercontent.com/your-org/your-repo/master/your_extension-v0.1.0-x86_64-apple-darwin.tar.gz ",
+            "sha256": "abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890"
+          },
+          "apple-darwin-aarch64": {
+            "url": "https://raw.githubusercontent.com/your-org/your-repo/master/your_extension-v0.1.0-aarch64-apple-darwin.tar.gz ",
+            "sha256": "abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The extension must be published as a tarball (`.tag.gz`), which contains a directory named `<EXTENSION_NAME>-v<EXTENSION_VERSION>-<CPU_ARCHITECTURE>-<PLATFORM>`, which contains a binary named `<EXTENSION_NAME>`, here is minimal example:
+
+```bash
+mkdir test_extension-v0.1.0-x86_64-unknown-linux-gnu
+cat > test_extension-v0.1.0-x86_64-unknown-linux-gnu/test_extension << "EOF"
+#!/usr/bin/env bash
+
+echo Hello, World!
+EOF
+tar -czf test_extension-v0.1.0-x86_64-unknown-linux-gnu.tar.gz test_extension-v0.1.0-x86_64-unknown-linux-gnu
+# you'll also need to generate a sha256 hash of the tarball and put it into the registry file
+shasum -a 256 test_extension-v0.1.0-x86_64-unknown-linux-gnu.tar.gz 
+```
```

Comment:
This is a little too much to put in the dfx changelog.  A better place for it would be a document in https://github.com/dfinity/dfx-extensions/docs which describes how to set up your own extension registry and publish extensions from it.  Then, we can link to that document from here.